`utils.rollbackHeight` rolls back to wrong height

**Describe the bug**
Following testcode:

```js
    // rollback to first keyblock
    await utils.rollbackHeight(aeSdk, 1);
    assert.equal(await aeSdk.getHeight(), 1);
```

Fails with:

```sh
AssertionError: expected 2 to equal 1
      + expected - actual

      -2
      +1
```

**Expected behavior**
I expect the assertion to pass, currently it fails.

**Please tell us about your environment:**

  - AEproject version: `v4.8.2`
  - Devmode version: `v0.5.2`

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
     "test": "mocha ./tests/**/*.js --timeout 0 --exit",
     "prepublishOnly": "npm run build",
     "copy-files": "cp -r ./src/init/artifacts ./.build/cjs/src/init/ && cp -r ./src/init/artifacts ./.build/esm/src/init/ && cp -r ./src/init/update-artifacts ./.build/cjs/src/init/ && cp -r ./src/init/update-artifacts ./.build/esm/src/init/",
-    "link:local": "npm run build:clean && npm run build && npm link"
+    "link:local": "npm uninstall -g @aeternity/aeproject && npm run build:clean && npm run build && npm link"
   },
   "license": "ISC",
   "engines": {

--- a/src/lib/utils.ts
+++ b/src/lib/utils.ts
@@ -75,6 +75,7 @@ let snapshotHeight = -1;
 
 export const createSnapshot = async (aeSdk) => {
   snapshotHeight = await aeSdk.getHeight();
+  await awaitKeyBlocks(aeSdk, 1);
 };
 
 export const rollbackHeight = async (aeSdk, height) => {

--- a/src/lib/utils.ts
+++ b/src/lib/utils.ts
@@ -81,7 +81,6 @@ export const rollbackHeight = async (aeSdk, height) => {
   const currentBlockHeight = await aeSdk.getHeight();
   if (currentBlockHeight > height) {
     await get(`http://localhost:3001/rollback?height=${height}`);
-    await awaitKeyBlocks(aeSdk, 1);
   }
 };
 

--- a/tests/cli.js
+++ b/tests/cli.js
@@ -1,29 +1,19 @@
 const path = require('path');
-const { exec: execP } = require('promisify-child-process');
-const fs = require('fs');
 const chai = require('chai');
 const chaiFiles = require('chai-files');
 
 const { isEnvRunning } = require('../src/env/env');
 const { version } = require('../package.json');
 const { print } = require("../src/utils/utils");
+const { exec, cwd, prepareLocal, cleanLocal, linkLocalLib } = require("./util");
 
 chai.use(chaiFiles);
 const { assert } = chai;
 const { file } = chaiFiles;
 
-const cwd = path.join(process.cwd(), '.testdir');
-const exec = (cmd, options) => execP(`${fs.existsSync('~/.profile') ? '. ~/.profile;' : ''}${cmd}`, options);
-
-describe('Happy Path', () => {
-  before(async () => {
-    await exec('npm run link:local');
-    if (!fs.existsSync(cwd)) fs.mkdirSync(cwd);
-  });
-
-  after(() => {
-    fs.rmSync(cwd, { recursive: true });
-  });
+describe('command line usage', () => {
+  before(async () => await prepareLocal());
+  after(() => cleanLocal());
 
   it('help', async () => {
     const res = await exec('aeproject help', { cwd });
@@ -43,7 +33,7 @@ describe('Happy Path', () => {
       assert.equal(res.code, 0);
 
       // link to use local aeproject utils
-      await exec('npm link @aeternity/aeproject', { cwd: folder ? path.join(cwd, folder) : cwd });
+      await linkLocalLib(folder)
 
       assert.exists(file(path.join(cwd, '.gitignore')));
       assert.exists(file(path.join(cwd, 'docker-compose.yml')));

--- a/tests/lib.js
+++ b/tests/lib.js
@@ -1,0 +1,59 @@
+const chai = require('chai');
+const chaiFiles = require('chai-files');
+const {utils} = require('@aeternity/aeproject');
+
+const {exec, cwd, prepareLocal, cleanLocal, linkLocalLib} = require("./util");
+const {assert} = require("chai");
+const {generateKeyPair} = require("@aeternity/aepp-sdk");
+
+chai.use(chaiFiles);
+
+describe('library usage', () => {
+    let aeSdk;
+
+    before(async () => {
+        await prepareLocal()
+        await exec('aeproject init', {cwd});
+        await linkLocalLib();
+
+        await exec('aeproject env', {cwd});
+
+        aeSdk = utils.getSdk();
+    });
+
+    after(async () => {
+        await exec('aeproject env --stop', {cwd});
+        cleanLocal()
+    });
+
+    it('await key blocks', async () => {
+        const height = await aeSdk.getHeight();
+        await utils.awaitKeyBlocks(aeSdk, 5);
+        assert.equal(await aeSdk.getHeight(), height + 5);
+    });
+
+    it('rollback height', async () => {
+        const height = await aeSdk.getHeight();
+
+        const {publicKey} = generateKeyPair();
+        await aeSdk.spend(100, publicKey)
+        assert.equal(await aeSdk.getBalance(publicKey), 100);
+
+        await utils.rollbackHeight(aeSdk, height);
+        assert.equal(await aeSdk.getHeight(), height);
+        assert.equal(await aeSdk.getBalance(publicKey), 0);
+    });
+
+    it('rollback snapshot', async () => {
+        const height = await aeSdk.getHeight();
+        await utils.createSnapshot(aeSdk);
+
+        const {publicKey} = generateKeyPair();
+        await aeSdk.spend(100, publicKey)
+        assert.equal(await aeSdk.getBalance(publicKey), 100);
+
+        await utils.rollbackSnapshot(aeSdk);
+        assert.equal(await aeSdk.getHeight(), height);
+        assert.equal(await aeSdk.getBalance(publicKey), 0);
+    });
+});

--- a/tests/util.js
+++ b/tests/util.js
@@ -1,0 +1,31 @@
+const path = require("path");
+
+const {exec: execP} = require('promisify-child-process');
+const fs = require('fs');
+
+const cwd = path.join(process.cwd(), '.testdir');
+
+async function exec(cmd, options) {
+    return execP(`${fs.existsSync('~/.profile') ? '. ~/.profile;' : ''}${cmd}`, options);
+}
+
+async function prepareLocal() {
+    await exec('npm run link:local');
+    if (!fs.existsSync(cwd)) fs.mkdirSync(cwd);
+}
+
+function cleanLocal() {
+    fs.rmSync(cwd, {recursive: true});
+}
+
+async function linkLocalLib(folder) {
+    await exec('npm link @aeternity/aeproject', {cwd: folder ? path.join(cwd, folder) : cwd});
+}
+
+module.exports = {
+    exec,
+    cwd,
+    prepareLocal,
+    cleanLocal,
+    linkLocalLib
+}

